lib/parsers/yarn-lock-parse.ts

```diff
@@ -87,33 +87,18 @@ export class YarnLockParser implements LockfileParser {
     }
 
     const topLevelDeps: Dep[] = getTopLevelDeps(manifestFile, includeDev);
-
     // asked to process empty deps
     if (_.isEmpty(manifestFile.dependencies) && !includeDev) {
       return depTree;
     }
 
-    for (const dep of topLevelDeps) {
-      if (/^file:/.test(dep.version)) {
-        depTree.dependencies[dep.name] = createDepTreeDepFromDep(dep);
-      } else {
-        depTree.dependencies[dep.name] = await this.buildSubTree(yarnLock, createDepTreeDepFromDep(dep), strict);
-      }
-      this.treeSize++;
-
-      if (this.treeSize % this.eventLoopSpinRate === 0) {
-        // Spin event loop every X dependencies.
-        await setImmediatePromise();
-      }
-    }
-
+    await Promise.all(topLevelDeps.map((dep) => this.resolveDep(dep, depTree, yarnLock, strict)));
```

Comment:
Maybe `pMap` with a limited concurrency here? Otherwise the Mac amount of promises on the event loop is not really bound.

Reply:
Is it safe to mutate this.treeSize concurrently?

Reply:
all the tests pass and it returns the exact same tree as before (as asserted by the tests), it only processes top level deps concurrently for now. this.treeSize is a class variable so should be no problem.